File: 0060-permutation-sequence/0060-permutation-sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> numbers;
        int factorial = 1;
        for (int i = 1; i <= n; ++i) {
            numbers.push_back(i);
            factorial *= i;
        }
        
        k--; // Convert k to zero-based index
        string result;
        
        for (int i = 0; i < n; ++i) {
            factorial /= (n - i);
            int index = k / factorial;
            result += to_string(numbers[index]);
            numbers.erase(numbers.begin() + index);
            k %= factorial;
        }
        
        return result;
    }
};

File: 0060-permutation-sequence/0060-permutation-sequence.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public String getPermutation(int n, int k) {
        List<Integer> numbers = new ArrayList<>();
        int factorial = 1;
        for (int i = 1; i <= n; i++) {
            numbers.add(i);
            factorial *= i;
        }
        
        k--; // Convert k to zero-based index
        StringBuilder result = new StringBuilder();
        
        for (int i = 0; i < n; i++) {
            factorial /= (n - i);
            int index = k / factorial;
            result.append(numbers.get(index));
            numbers.remove(index);
            k %= factorial;
        }
        
        return result.toString();
    }
}
